thrift_rpc/recorder.go

```diff
@@ -451,19 +466,25 @@ func (r *Recorder) reportLoop() {
 	}
 
 	tickerChan := time.Tick(minReportingPeriod)
-	for range tickerChan {
-		r.maybeLogInfof("reporting alarm fired")
-
-		// Kill the reportLoop() if we've been disabled.
-		r.lock.Lock()
-		if r.disabled {
+	for {
+		select {
+		case <-tickerChan:
+			r.maybeLogInfof("reporting alarm fired")
+
+			// Kill the reportLoop() if we've been disabled.
+			r.lock.Lock()
+			if r.disabled {
+				r.lock.Unlock()
+				break
+			}
 			r.lock.Unlock()
-			break
-		}
-		r.lock.Unlock()
 
-		if r.shouldFlush() {
-			r.Flush()
+			if r.shouldFlush() {
+				r.Flush()
+			}
+
+		case <-closech:
+			return
```

Comment:
Just to be clear: if there is a need for the thrift tracer to have a new lease on life, I will create a new PR focused on it. We do not expect users to be writing new code with it and will be EOL the thrift option shortly. If you are planing on writing new thrift based code, please let me know! (And also, please let me know why the gRPC option would not work for you.)